gtda/diagrams/representations.py

```diff
@@ -696,10 +701,16 @@ def plot(self, Xt, sample=0, homology_dimension_idx=0, colorscale="blues",
 
         """
         check_is_fitted(self)
-        x = self.samplings_[self.homology_dimensions_[homology_dimension_idx]]
+        homology_dimension = self.homology_dimensions_[homology_dimension_idx]
+        if homology_dimension != np.inf:
+            homology_dimension = int(homology_dimension)
+        x = self.samplings_[homology_dimension]
```

Comment:
Does that work with `homology_dimension == np.inf`?

Reply:
I don't recall when it is that this condition is satisfied.

Reply:
`ForgetDimension` will change all hom dims to `np.inf`. The screenshots above show precisely this scenario.